contracts/token/ERC721/extensions/ERC721Consecutive.sol

```diff
@@ -141,8 +141,15 @@ abstract contract ERC721Consecutive is IERC2309, ERC721 {
         super._afterTokenTransfer(from, to, firstTokenId, batchSize);
     }
 
+    /**
+     * @dev Used to offset the first token id in {_totalConsecutiveSupply}
+     */
+    function _firstConsecutiveId() internal view virtual returns (uint96) {
+        return 0;
+    }
+
     function _totalConsecutiveSupply() private view returns (uint96) {
         (bool exists, uint96 latestId, ) = _sequentialOwnership.latestCheckpoint();
-        return exists ? latestId + 1 : 0;
+        return exists ? latestId + 1 : _firstConsecutiveId();
```

Comment:
Would requiring that the tokenId be greater than the last batch minted one in the `_mint` function be a viable option? It might be too inefficient... 
```solidity
    function _mint(address to, uint256 tokenId) internal virtual override {
        require(Address.isContract(address(this)), "ERC721Consecutive: can't mint during construction");
        (, uint96 latestId, ) = _sequentialOwnership.latestCheckpoint();
        require(tokenId > latestId, "ERC721Consecutive: token id must be greater than latest id");
        super._mint(to, tokenId);
    }
```

Reply:
That is something we don't want, because it adds 2 (cold) sload to the mint operation, which we want to keep as cheap as possible.

Also, if we do that, it would be impossible to `_mint` a token that was first minted as part of a batch then burnt.

Reply:
note that mint already check if the tokenId has an owner, and revert if that is the case.

Reply:
I see, I'm still wrapping my head around the internals of this but I think I got it.